kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/test/interceptors/BeforeSpecListenerInterceptor.kt

```diff
@@ -42,9 +45,15 @@ internal class BeforeSpecListenerInterceptor(
             state == null -> suspend { test(testCase, scope) }
             state.success.contains(testCase.spec) -> suspend { test(testCase, scope) }
             state.failed.contains(testCase.spec) -> suspend { TestResult.Ignored("Skipped due to beforeSpec failure") }
-            else -> SpecExtensions(registry)
-               .beforeSpec(testCase.spec)
-               .fold(
+            else -> {
+
+               // beforeSpec needs to execute on the spec level coroutine, but here we are in
+               // the test executor pipline. We are doing the beforeSpec here and not in the spec pipeline
+               // because we only want to run it when we see an enabled test.
+               val context = scope.coroutineContext[SpecContextElement] ?: error("No SpecContextElement")
+               CoroutineScope(context.context).async {
+                  SpecExtensions(registry).beforeSpec(testCase.spec)
+               }.await().fold(
```

Comment:
```suggestion
            else -> {
               // beforeSpec needs to execute on the spec level coroutine, but here we are in
               // the test executor pipline. We are doing the beforeSpec here and not in the spec pipeline
               // because we only want to run it when we see an enabled test.
               val specContext = scope.coroutineContext[SpecContextElement]?.context ?: error("No SpecContextElement")
               withContext(specContext) {
                  SpecExtensions(registry).beforeSpec(testCase.spec)
               }.fold(
```

This is equivalent yet more idiomatic.

Reply:
([IntelliJ failed to properly support a multiline suggestion.](https://youtrack.jetbrains.com/issue/IJPL-73076))